packages/headless/src/modules/product/services/index.ts

```diff
@@ -181,4 +184,31 @@ export class ProductService {
       brands: new Array(...brands)
     }, HttpStatus.OK);
   }
+
+  async getCustomizedProducts(condition: GetCustomizedProductsQuery): Promise<GetCustomizedProductsResponse> {
+    const query = { isHomePageProductsTags: true };
+    const Tags = await this.productRepo.getTags(query);
+
+    let tagName = Tags.map((tag) => { return tag.name });
+    let productObject = {};
+    let products = await Promise.all(tagName.map(async (name) => {
+      const product = { [name]: await this.getProductsByTags(name, condition)}
+      productObject = Object.assign(productObject,product);
+      return product;
```

Comment:
no, you won't return all tags product in a single api call.
rather front end will call the same api by passing a single tag each time, and you will return the products of a single tag each time.
so if in home page there are 5 different tags then front end will call this api 5 times.